pkg/networkserver/downlink.go

```diff
@@ -642,21 +642,37 @@ type downlinkPath struct {
 	*ttnpb.DownlinkPath
 }
 
-func buildMetadataComparator(mds []*ttnpb.MACState_UplinkMessage_RxMetadata) func(int, int) bool {
-	invalidMD := func(k int) bool { return mds[k].Snr == 0.0 || mds[k].ChannelRssi == 0.0 }
-	wantedRSSI := func(k int) float32 { return lora.AdjustedRSSI(mds[k].ChannelRssi, mds[k].Snr) }
-	return func(i, j int) bool {
-		lhsInvalid, rhsInvalid := invalidMD(i), invalidMD(j)
-		if lhsInvalid {
-			return lhsInvalid == rhsInvalid
+func buildMetadataComparator(
+	settings *ttnpb.MACState_UplinkMessage_TxSettings, mds []*ttnpb.MACState_UplinkMessage_RxMetadata,
+) func(int, int) bool {
+	build := func(invalid func(int) bool, measure func(int) float32) func(i, j int) bool {
+		return func(i, j int) bool {
+			lhsInvalid, rhsInvalid := invalid(i), invalid(j)
+			if lhsInvalid {
+				return lhsInvalid == rhsInvalid
+			}
+			return measure(i) >= measure(j)
 		}
-		return wantedRSSI(i) >= wantedRSSI(j)
+	}
+	switch settings.DataRate.Modulation.(type) {
+	case *ttnpb.DataRate_Lora:
+		return build(
+			func(k int) bool { return mds[k].Snr == 0.0 || mds[k].ChannelRssi == 0.0 },
```

Comment:
An SNR of `0.0` is not invalid

Reply:
This check has always been here since we first added this filtering (https://github.com/TheThingsNetwork/lorawan-stack/pull/4812).

I ran some queries on TTSCE `eu1` and indeed does occur (about 1% of the time). Removed in https://github.com/TheThingsNetwork/lorawan-stack/pull/5877.